knimin/handlers/pm_normalize.py

```diff
@@ -0,0 +1,86 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The LabAdmin Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from tornado.web import authenticated
+from datetime import date
+
+from knimin.handlers.base import BaseHandler
+from knimin.handlers.access_decorators import set_access
+from knimin import db
+from knimin.lib.parse import parse_plate_reader_output
+from knimin.lib.format import format_normalization_echo_pick_list
+from knimin.lib.shotgun import compute_shotgun_normalization_values
+
+
+@set_access(['Admin'])
+class PMNormalizeHandler(BaseHandler):
+    @authenticated
+    def get(self):
+        plate_id = self.get_argument('plate')
+
+        plate = db.read_shotgun_plate(plate_id)
+        plate['created_on'] = plate['created_on'].isoformat(sep=' ')
+
+        condensed = plate['condensed_plates']
+        plate['condensed_plates'] = []
+        for pid, pos in condensed:
+            p = db.read_dna_plate(pid)
+            p['created_on'] = p['created_on'].isoformat(sep=' ')
+            plate['condensed_plates'].append((pos, p))
+
+        self.render("pm_normalize.html", plate=plate,
+                    plate_readers=db.get_property_options("plate_reader"),
+                    echos=db.get_property_options("echo"))
+
+    @authenticated
+    def post(self):
+        plate_id = self.get_argument("plate_id")
+        input_vol = self.get_argument("pm-volume")
+        input_dna = self.get_argument("pm-dna-input")
+        upload_type = self.get_argument("upload-select")
+        plate_reader = self.get_argument("plate-reader")
+        echo = self.get_argument("echo")
+
+        if upload_type == 'Single file':
+            qubit_assay = self.request.files['single-plate-fp'][0]['body']
+            db.quantify_shotgun_plate()
+            dna_conc = parse_plate_reader_output(qubit_assay)
+            cond_dna_conc = None
+        else:
+            qubit_assay_0 = self.request.files['plate-0-fp'][0]['body']
```

Comment:
Maybe not everyone's favorite. But this thing can be written as (just for reference):
```
for i in range(4):
    vars()['qubit_assay_%s' % i] = self.request.files['plate-%s-fp' % i][i]['body']
    vars()['dna_conc_%s' % i] = parse_plate_reader_output(vars()['qubit_assay_%s' % i])
```